test.py

```diff
@@ -1,30 +1,53 @@
+from crawler import config
 from crawler.rss import RSSCrawler
 from crawler.github import GithubCrawler
-from crawler.config import RSSConfig as conf
-import json, sys
+from crawler.config import RSSConfig as rss_conf
+import unittest
+import requests
 
-def test_rss():
-    # RSS test
-    # TODO prevent from unappropriate input
-    newspaper = input("[+] Newspaper name: ")
-    rss_result = RSSCrawler(newspaper)
-    print(json.dumps(rss_result, indent=4))
+class TestRSSCrawler(unittest.TestCase):
+    """
+    Testcase for RSSCrawler.
+    """
+    def setUp(self):
+        self.feed_list = rss_conf.links.keys()
+        self.props = rss_conf.properties
 
-def test_github():
-    # Github test
-    # TODO prevent from fake username
-    username = input("[+] GitHub Username: ")
-    github_result = GithubCrawler(username)
-    print(json.dumps(github_result, indent=4))
+    def test_links_are_available(self):
+        """
+        Test for news RSS feed links.
+        This test passes when all rss feed links are available.
+        """
+        for newspaper in self.feed_list:
+            response = requests.get(rss_conf.links[newspaper], headers=config.req_header)
+            response.raise_for_status()
+    
+    def test_rss_crawl(self):
+        """
+        Test for crawling rss feeds.
+        This test passes when all feed returns one or more articles.
+        """
+        for newspaper in self.feed_list:
+            rss_result = RSSCrawler(newspaper)
+            self.assertTrue(len(rss_result) >= 1)
 
-def main():
-    print("[+] Type 1 for RSS, 2 for Github. Other to exit.")
-    choose = input("[?] Choose > ")
-    if choose == "1": test_rss()
-    elif choose == "2": test_github()
-    else: sys.exit(0)
+class TestGitHubCrawler(unittest.TestCase):
+    """
+    Testcase for GithubCrawler.
+    """
+    def test_getting_profile(self):
+        """
+        Test for crawling Github profiles.
+        This test passes when nickname and repositories are successfully parsed.
+        """
+        # TODO Generalize test case
+        profile = GithubCrawler("harrydrippin")
```

Comment:
Same as above. Set up a fake response X for GithubCrawler, and the expected parsed result Y, test if profile[X] == Y.

Reply:
I'll refactor all of them.